app/bundles/LeadBundle/Segment/Decorator/DateDecorator.php

```diff
@@ -36,4 +37,46 @@ public function getDefaultDate()
     {
         return new DateTimeHelper('midnight today', null, 'local');
     }
+
+    /**
+     * @param ContactSegmentFilterCrate $contactSegmentFilterCrate
+     *
+     * @return string
+     */
+    public function getQueryType(ContactSegmentFilterCrate $contactSegmentFilterCrate)
+    {
+        if ($contactSegmentFilterCrate->isCompanyType()) {
```

Comment:
Can we create a new decorator for this case? I will prevent this if + could solve issue with BC break.